lib/src/checkbox/checkbox.component.spec.ts

```diff
@@ -1,26 +1,284 @@
+            fixture.detectChanges();
+
+            expect(inputElement.value).toBe('basic_checkbox');
+        });
     });
 
-    it('should create', () => {
-        expect(component).toBeTruthy();
+    describe('with form control', () => {
+        let checkboxDebugElement: DebugElement;
+        let checkboxInstance: CheckboxComponent;
+        let testComponent: CheckboxWithFormControlComponent;
+        let inputElement: HTMLInputElement;
+
+        beforeEach(() => {
+            fixture = TestBed.createComponent(CheckboxWithFormControlComponent);
+            fixture.detectChanges();
+
+            checkboxDebugElement = fixture.debugElement.query(By.directive(CheckboxComponent));
+            checkboxInstance = checkboxDebugElement.componentInstance;
+            testComponent = fixture.debugElement.componentInstance;
+            inputElement = <HTMLInputElement>checkboxDebugElement.nativeElement.querySelector('input');
+        });
+
+        it('should toggle the disabled state', () => {
+            expect(checkboxInstance.disabled).toBe(false);
+
+            testComponent.formControl.disable();
+            fixture.detectChanges();
+
+            expect(checkboxInstance.disabled).toBe(true);
+            expect(inputElement.disabled).toBe(true);
+
+            testComponent.formControl.enable();
+            fixture.detectChanges();
+
+            expect(checkboxInstance.disabled).toBe(false);
+            expect(inputElement.disabled).toBe(false);
+        });
+    });
+
+    describe('with required ngModel', () => {
+        let checkboxInstance: CheckboxComponent;
+        let inputElement: HTMLInputElement;
+        let testComponent: CheckboxWithNgModelComponent;
+
+        beforeEach(() => {
+            fixture = TestBed.createComponent(CheckboxWithNgModelComponent);
+            fixture.detectChanges();
+
+            let checkboxDebugElement = fixture.debugElement.query(By.directive(CheckboxComponent));
+            let checkboxNativeElement = checkboxDebugElement.nativeElement;
+            testComponent = fixture.debugElement.componentInstance;
+            checkboxInstance = checkboxDebugElement.componentInstance;
+            inputElement = <HTMLInputElement>checkboxNativeElement.querySelector('input');
+        });
+
+        it('should validate with RequiredTrue validator', () => {
+            let checkboxElement = fixture.debugElement.query(By.directive(CheckboxComponent));
+            let ngModel = checkboxElement.injector.get<NgModel>(NgModel);
+
+            testComponent.isRequired = true;
+            inputElement.click();
+            fixture.detectChanges();
+
+            expect(checkboxInstance.checked).toBe(true);
+            expect(ngModel.valid).toBe(true);
+
+            inputElement.click();
+            fixture.detectChanges();
+
+            expect(checkboxInstance.checked).toBe(false);
+            expect(ngModel.valid).toBe(false);
+        });
     });
 });
+
+@Component({
```

Comment:
If this is moved to the top, you could remove the tslint ignore

Reply:
I want to keep those at the bottom so it's easier to read the unit tests. The tests are the main focal point not those components

Reply:
If thats the case, maybe we should remove this from the tslint rules.